src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricExporter.cs

```diff
@@ -127,11 +136,12 @@ public override ExportResult Export(in Batch<Metric> metrics)
                 }
                 finally
                 {
-                    request.Return();
+                    ArrayPool<List<OtlpMetrics.Metric>>.Shared.Return(perMetricBatches);
+                    ArrayPool<Metric>.Shared.Return(meterNames);
+                    request?.Return();
                 }
             }
-
-            Console.WriteLine($"\nSent {num_msg} messages totaling {wire_size / 1024.0f / 1024.0f} MB of data");
+            while (request != null);
```

Comment:
This logic doesn't work.

It's still using the same request object across batches, even worse is that it returns the request and then re-uses it. I'm surprised unit tests passed with this change.

We can keep the `while()` loop, Upon successful sending of the request, it can be set to `null` before the start of the next iteration of the while loop.

Reply:
This has been addressed.